### PR TITLE
implemented describe_na_values and tests

### LINKED ISSUE #30: implement describe_na_values

### LINKED ISSUE #32: implement test_describe_na_values

### PR DESCRIPTION
**Pull request by v5y8**
closes #30 
closes #32 
added dependencies and dev-dependencies

### CODE REVIEW

File: eda_analysis/eda_analysis.py [hunk 1]
```diff
@@ -46,11 +49,43 @@ def describe_na_values(dataframe):
...
+                                           }),
+    >>> describe_na_variable(na_numerical_dataframe)
+    numpy.ndarray([1, 1],
+                  [1, 1],
+                  [0, 1])
+
     '''
-    pass    
+    bool_array = dataframe.isna()
+    return np.array([[0 if val else 1 for val in bool_array[col]] for col in bool_array.columns])
```
Comment (sgauravm) [inline eda_analysis/eda_analysis.py]:
```suggestion
    na_val = np.array([[0 if val else 1 for val in bool_array[col]] for col in bool_array.columns])
    return pd.Series(data = np.sum(na_val!=1,axis=1),index = list(dataframe.columns))
```
OPTIONAL SUGGESTION:I really liked the one line logic you wrote for this function.But the array itself is not self explanatory. It is the same as finding the number of na values manually. You can return a series as suggested above. I tested that in my local by copying your function. You can use the suggested chunk or write your own logic,

File: eda_analysis/eda_analysis.py [hunk 2]
```diff
@@ -46,11 +49,43 @@ def describe_na_values(dataframe):
     numpy.ndarray
         A 2d Numpy Array of 1's and 0's, corresponding to the value of each entry in the dataframe.
         0 represents an NA value, 1 represents a non-NA value.
```
Comment (sgauravm) [inline eda_analysis/eda_analysis.py:53]: You can modify this according to the changes suggested in return of the function. This should nbe either pandas.Series or pandas.DataFrame

File: eda_analysis/eda_analysis.py [hunk 3]
```diff
@@ -46,11 +49,43 @@ def describe_na_values(dataframe):
...
+                                           "col_1": [0, 2],
+                                           "col_2": [0.5, 0.1],
+                                           "col_3": [np.nan, "b"]
+                                           }),
+    >>> describe_na_variable(na_numerical_dataframe)
+    numpy.ndarray([1, 1],
+                  [1, 1],
+                  [0, 1])
+
     '''
```
Comment (sgauravm) [inline eda_analysis/eda_analysis.py:85]: Include some defensive  coding like these :
```
    if not isinstance(dataframe, pd.DataFrame):
        raise Exception("The value of the argument 'dataframe' must be of type 'pandas.DataFrame'")
    
    if not isinstance(cat_vars,list) or not all(isinstance(x, str) for x in cat_vars):
        raise Exception("The value of the argument 'cat_vars' must be a list of strings")
```
Defensive coding is one of the requirement and even your test case should also have some negative testcase for these

File: tests/test_eda_analysis.py [hunk 4]
```diff
@@ -1 +1,33 @@
...
+
+    assert isinstance(eda_analysis.describe_na_values(na_numerical_dataframe), np.ndarray)
+    assert np.array_equiv(eda_analysis.describe_na_values(na_numerical_dataframe), np.array([[1, 1],
+                                                                                             [0, 1],
+                                                                                             [1, 1]]))
+
+    assert isinstance(eda_analysis.describe_na_values(na_categorical_dataframe), np.ndarray)
+    assert np.array_equiv(eda_analysis.describe_na_values(na_categorical_dataframe), np.array([[1, 1],
+                                                                                               [1, 1],
+                                                                                               [0, 1]]))
```
Comment (sgauravm) [inline tests/test_eda_analysis.py]: If you are making the changes suggested in your main function, you might have to modify the test cases. Even if you are not making any changes in your main function then just add some test case for the exceptions case (Defensive coding).

**Review by sgauravm — CHANGES_REQUESTED** [review]

File: eda_analysis/eda_analysis.py (hunk 1 above)
Comment (sgauravm) [reply eda_analysis/eda_analysis.py]: You can leave it for now as you will have to make changes in test file R function and its test. I am making it as an optional change

**Comment by v5y8** [comment]
@sgauravm I've made the changes; please take a look

**Review by sgauravm — APPROVED** [review]
Looks good . You can merge now

**Pull request merged**